code/modules/clothing/suits/marine_armor.dm

```diff
@@ -362,75 +365,99 @@ var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200),
 
 	var/list/details = list()
 	var/dose_administered = null
-	var/tricordrazine = REAGENTS_OVERDOSE - (wearer.reagents.get_reagent_amount("tricordrazine") + 0.5)
+	var/tricordrazine = CLAMP(REAGENTS_OVERDOSE - (wearer.reagents.get_reagent_amount("tricordrazine") + 0.5),0,REAGENTS_OVERDOSE - 0.5)
 
 	if(wearer.getFireLoss() > B18_automed_damage && !B18_burn_cooldown)
-		var/dermaline = REAGENTS_OVERDOSE*0.5 - (wearer.reagents.get_reagent_amount("dermaline") + 0.5)
-		var/kelotane = REAGENTS_OVERDOSE - (wearer.reagents.get_reagent_amount("kelotane") + 0.5)
-		if(dermaline < REAGENTS_OVERDOSE*0.5)
-			wearer.reagents.add_reagent("dermaline",REAGENTS_OVERDOSE*0.5 - dermaline + 0.5)
-		if(kelotane < REAGENTS_OVERDOSE)
-			wearer.reagents.add_reagent("kelotane",REAGENTS_OVERDOSE - kelotane + 0.5)
-		if(tricordrazine < REAGENTS_OVERDOSE)
-			wearer.reagents.add_reagent("tricordrazine",REAGENTS_OVERDOSE - tricordrazine + 0.5)
-		details +=("Significant tissue burns detected. Restorative injection administered. <b>Dosage: Dermaline: [dermaline]U | Kelotane: [kelotane]U | Tricordrazine: [tricordrazine]U</b></br>")
-		B18_burn_cooldown = world.time + B18_CHEM_COOLDOWN
-		handle_chem_cooldown(B18_BURN_CODE)
-		dose_administered = TRUE
+		var/dermaline = CLAMP(REAGENTS_OVERDOSE*0.5 - (wearer.reagents.get_reagent_amount("dermaline") + 0.5),0,REAGENTS_OVERDOSE*0.5 - 0.5)
```

Comment:
The + 0. 5 is just useless now that it clamps at the OD threshold - 0.5, remove it.

Reply:
Nah. If someone is at the OD threshold, they can potentially suffer a tick of OD effects per my testing which is why I added it; this is a safety buffer.